internal/hierarchy/hierarchy.go

```diff
@@ -48,6 +48,22 @@ func (p pathValue) Equal(o any) bool {
 	return true
 }
 
+// GetEndpoint returns the hierarchical path of endpoint.
+func GetEndpoint(endpoint resolver.Endpoint) []string {
+	attrs := endpoint.Attributes
+	if attrs == nil {
```

Comment:
I think this function can just be:

```go
path, _ := endpoint.Attributes.Value(pathKey).(pathValue)
return path
```

Reply:
So the reason is because we're guaranteed this to be set?

So programmer error vs. simplicity? Changed.

Reply:
No, because they are actually equivalent.  :)